File: separate_the_numbers.py
import sys

def isBeautiful(s):
    for digits in range(1, int(len(strg)/2 + 1), 1):
            substring = strg[0:digits]
            start = int(substring)
            ret = int(substring)
            while len(substring) < len(strg):
                start = start + 1
                substring = substring + str(start)
            if len(substring) is len(strg):
                if substring == strg:
                    return ret
    return 0

no_of_strings = int(input())
for nos in range(no_of_strings):
    strg = input()
    if strg[0] is 0 and len(strg) is 1:
        print("NO")
        break
    ret = isBeautiful("99100")
    if ret is not 0:
        print("YES", ret)
    else:
        print("NO")
    

File: separate_the_numbers.cpp
#include<iostream>
#include<string>
using namespace std;

bool isValidString(string s){
    if(s.size()<=1 || s[0] == '0')
        return false;
    return true;
}

long int isBeautiful(string s){
    if(!isValidString)
        return 0;
    
    for(int noDigits=1;noDigits<=s.length()/2;++noDigits){
        string substring = s.substr(0,noDigits);
        long int nextDigit = stol(substring);
        long int ret = nextDigit;

        while(substring.size() < s.size()){
            substring += to_string(++nextDigit); 
        }

        if(substring.size() == s.size() && substring == s){
            return ret;
        }
    }
    
    return 0;
}

int main(){
    int no_of_strings;
    cin>>no_of_strings;
    for(int i=0;i<no_of_strings;++i){
        string s;
        cin>>s;
        long int ret = isBeautiful(s);
        if(ret == 0)
            cout<<"NO\n";
        else
            cout<<"YES "<<ret<<"\n";
    }
}